Classes/Middleware/TopwireRendering.php

```diff
@@ -41,24 +42,26 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
 
     private function validateContentType(ServerRequestInterface $request, ResponseInterface $response): ResponseInterface
     {
-        if (!$request->hasHeader('Turbo-Frame')
+        $contentTypeHeader = $response->getHeaderLine('Content-Type');
+        $isStreamResponseAllowed = str_contains($request->getHeaderLine('Accept'), self::turboStreamContentType);
+        if ($contentTypeHeader === ''
             || $response->getStatusCode() !== 200
-            || !$response->hasHeader('Content-Type')
+            || !$request->hasHeader('Turbo-Frame')
+            || str_starts_with($contentTypeHeader, self::defaultContentType)
+            || ($isStreamResponseAllowed && str_starts_with($contentTypeHeader, self::turboStreamContentType))
         ) {
             return $response;
         }
-        $contentTypeHeader = $response->getHeaderLine('Content-Type');
-        if (!str_starts_with($contentTypeHeader, self::defaultContentType)) {
-            throw new InvalidContentType(
-                sprintf(
-                    'Turbo requests must return content/type "text/html", got "%s". '
-                    . 'Maybe forgot to add data-turbo="false" attribute for links leading to this error? '
-                    . 'Alternatively you can rewrite the current URL to have a file extension',
-                    $contentTypeHeader
-                ),
-                1671308188
-            );
-        }
-        return $response;
+        throw new InvalidContentType(
+            sprintf(
+                'Turbo frame requests must return content/type "%s",%s got "%s". '
```

Comment:
```suggestion
                'Turbo frame requests must return content/type "%s"%s, got "%s". '
```

Reply:
I'm staring at the code and don't see what could be wrong. Could you elaborate?

Reply:
Ah, yeah, saw the comma issue, but was not sure you actually meant that detail ;)